crates/sui-config/src/lib.rs

```diff
@@ -0,0 +1,501 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use debug_ignore::DebugIgnore;
+use multiaddr::Multiaddr;
+use narwhal_config::Parameters as ConsensusParameters;
+use narwhal_config::{
+    Authority, Committee as ConsensusCommittee, PrimaryAddresses, Stake, WorkerAddresses,
+};
+use narwhal_crypto::ed25519::Ed25519PublicKey;
+use rand::rngs::OsRng;
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Serialize};
+use std::fs;
+use std::path::{Path, PathBuf};
+use sui_framework::DEFAULT_FRAMEWORK_PATH;
+use sui_types::base_types::{encode_bytes_hex, ObjectID, SuiAddress};
+use sui_types::committee::{Committee, EpochId};
+use sui_types::crypto::{get_key_pair_from_rng, KeyPair, PublicKeyBytes};
+use tracing::trace;
+
+pub mod utils;
+
+const DEFAULT_STAKE: usize = 1;
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct ValidatorConfig {
+    key_pair: KeyPair,
+    db_path: PathBuf,
+    network_address: Multiaddr,
+    metrics_address: Multiaddr,
+
+    consensus_config: ConsensuseConfig,
+    committee_config: CommitteeConfig,
+}
+
+impl Config for ValidatorConfig {}
+
+impl ValidatorConfig {
+    pub fn key_pair(&self) -> &KeyPair {
+        &self.key_pair
+    }
+
+    pub fn public_key(&self) -> PublicKeyBytes {
+        *self.key_pair.public_key_bytes()
+    }
+
+    pub fn sui_address(&self) -> SuiAddress {
+        SuiAddress::from(self.public_key())
+    }
+
+    pub fn db_path(&self) -> &Path {
+        &self.db_path
+    }
+
+    pub fn network_address(&self) -> &Multiaddr {
+        &self.network_address
+    }
+
+    pub fn consensus_config(&self) -> &ConsensuseConfig {
+        &self.consensus_config
+    }
+
+    pub fn committee_config(&self) -> &CommitteeConfig {
+        &self.committee_config
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct ConsensuseConfig {
+    consensus_address: Multiaddr,
+    consensus_db_path: PathBuf,
+
+    //TODO make narwhal config serializable
+    #[serde(skip_serializing)]
+    #[serde(default)]
+    narwhal_config: DebugIgnore<ConsensusParameters>,
```

Comment:
curious why `DebugIgnore`?

Reply:
ConsensusParameters doesn't impl Debug, I plan to fix it but need to do so in the narwhal repo